app/models/service_provider_request.rb

```diff
@@ -1,4 +1,6 @@
 class ServiceProviderRequest < ActiveRecord::Base
+  belongs_to :service_provider, foreign_key: :issuer, primary_key: :issuer
```

Comment:
Same here. I don't think we want this. Usually, a `belongs_to` in one model should correspond to a `has_many` in the other model. If we think we need this relationship in the app, then I think that would be a separate PR that changes how ServiceProviderRequest is defined.

I haven't compared the performance of both. Maybe it's worth visiting. Another thing to consider is that each `ServiceProviderRequest` that gets created (when someone comes to the IdP via an SP) gets deleted after the user is returned to the SP, so this data is short-lived.